### PR TITLE
Cross platform support

### PR DESCRIPTION
**Pull request by NSTCG**
* Added Virtual Joystick for mobile 
* Changed framework , to support AR
* 1 tap locomotion for cardboard and AR ( press and hold 1sec)
* Incremental speed on pc ( when pressing shift)
* Disabled controller mesh and related meshes on AR ,cardboard mobile and PC sessioans
* Restricted player vertical movement
* World's boundary is set to restric player movement outside it 
* Added inVR teleportation
* Added Full screen button
* 360 view  toggle for device orientation feature

### CODE REVIEW

**Comment by msub2** [comment]
Functionality looks good, but just from a code-formatting perspective I would appreciate it if you could:
- Clean up any lines that are just commented out and not going to be used for anything.
- Reduce the amount of white space, I see a lot of instances where there's 4 lines of whitespace for no real reason
- Make sure indentation is consistent, I see a lot of misaligned lines
- Make spacing consistent, (space after if/else, before curly braces, on each side of an equation operator, etc)

Let me know if you need more specific examples

**Comment by NSTCG** [comment]
Thanks for the feedback!
Formatted the code.

**Comment by Srile** [comment]
Looks good so far. Thank you for your work. I have some comments:

1. Please put each component in its own `.js` file, as that significantly improves readability, I would even suggest moving components dedicated to locomotion into a subfolder of /js/ named `locomotion`.

2. Are all the images comitted required to be in the project?

Thank you, I think after those are resolved, I can get on with merging.

File: js/auto-join-peer.js [hunk 1]
```diff
@@ -96,7 +97,7 @@ WL.registerComponent(
     // Host functions
     //
     host: function () {
-      const hostId = `host-${this.username}-${this.roomName}`;
```
Comment (msub2) [inline js/auto-join-peer.js:99]: Do not change this, this is required for the app to work.

File: js/auto-join-peer.js [hunk 2]
```diff
@@ -116,7 +117,7 @@ WL.registerComponent(
       this.peer = new Peer(hostId, peerObject);
   
       this.peer.on("open", this._onHostOpen.bind(this));
-      this.peer.on("error", this._onHostError.bind(this));
+      // this.peer.on("error", this._onHostError.bind(this));
```
Comment (msub2) [inline js/auto-join-peer.js:120]: Why is this commented out?

File: js/auto-join-peer.js [hunk 3]
```diff
@@ -238,7 +239,7 @@ WL.registerComponent(
       if (!id)
         return console.error("peer-manager: Connection id parameter missing");
       this.userId = `user-${this.username}_${id}`;
-      this.serverId = id;
```
Comment (msub2) [inline js/auto-join-peer.js:241]: Do not change this, this is required for the app to work.

File: js/auto-join-peer.js [hunk 4]
```diff
@@ -258,7 +259,7 @@ WL.registerComponent(
         this.peer = new Peer(this.userId, peerObject);
         this.peer.on("open", this._clientOnOpen.bind(this));
         this.peer.on("disconnected", this._onDisconnected.bind(this));
-        this.connectionId = id;
```
Comment (msub2) [inline js/auto-join-peer.js:261]: Do not change this, this is required for the app to work.

File: js/auto-join-peer.js [hunk 5]
```diff
@@ -17,7 +17,8 @@ WL.registerComponent(
     // Initialization
     //
     init: function () {
-      this.isLocalhost = window.location.hostname === 'localhost';
+      window.pc = this;
+      this.isLocalhost = false;
```
Comment (msub2) [inline js/auto-join-peer.js:21]: Why did you change this? You can circumvent this check locally by navigating to 127.0.0.1 instead of localhost

**Review by msub2 — CHANGES_REQUESTED** [review]
Please see my comments, I should've caught it the first time I looked at it but there are fields here that shouldn't be changed for this to work properly on the website

File: js/auto-join-peer.js (hunk 5 above)
Comment (Srile) [reply js/auto-join-peer.js:21]: There is stuff from local testing in there that I will be removing once the rest of the changes are in.

File: js/auto-join-peer.js (hunk 4 above)
Comment (Srile) [reply js/auto-join-peer.js:261]: This will also not be merged in.

File: js/auto-join-peer.js (hunk 3 above)
Comment (Srile) [reply js/auto-join-peer.js:241]: Will not be merged in.

File: js/auto-join-peer.js (hunk 2 above)
Comment (Srile) [reply js/auto-join-peer.js:120]: This causes an automatic join function on failed hosting, don't think we need this?

Comment (Srile) [reply js/auto-join-peer.js:120]: I removed that in `networking-testing`

File: js/auto-join-peer.js (hunk 1 above)
Comment (Srile) [reply js/auto-join-peer.js:99]: Yes, this will not be merged in

**Comment by NSTCG** [comment]
> Looks good so far. Thank you for your work. I have some comments:
> 
> 1. Please put each component in its own `.js` file, as that significantly improves readability, I would even suggest moving components dedicated to locomotion into a subfolder of /js/ named `locomotion`.
> 2. Are all the images comitted required to be in the project?
> 
> Thank you, I think after those are resolved, I can get on with merging.

Thanks @Srile  for the comments 
Put each component in its own .js file
made a dedicated folder for locomotion and added it project js path

as of commited images , just put that for reference for future , if we wanna switch color / design of controllers

**Comment by limbofeather** [comment]
@Srile @msub2 Are we good to merge or are there more changes that need to addressed?

**Comment by msub2** [comment]
@Srile still needs to push some changes I went over with him earlier

**Comment by limbofeather** [comment]
Ah ok, thanks for heads-up.

**Comment by Srile** [comment]
Just tested it locally, the joysticks don't seem to work properly. I can only rotate around the Y axis with the left thumbstick and nothing else. 

Ideally, I would move with the left joystick and have full y and x axis view with the right one. Can you fix that? @NSTCG

**Pull request merged**